### DISCUSSION TITLE
Есть пример создания Mini App на основе вайта?

### DISCUSSION THREAD

**Discussion by scffs** [VKCOM/VKUI#4551] [Q&A] [answered]
Всем привет, получалось у кого-нибудь сделать мини апп на вайте (react + vite)?

**Comment by SevereCloud** [comment, marked as answer]
Привет, в [быстром старте](https://vkcom.github.io/VKUI/#/QuickStart) описано как создавать приложение на VKUI. Для работы VK Mini Apps дополнительно необходимо добавить вызов [VKWebAppInit](https://dev.vk.com/bridge/VKWebAppInit)

### Шаг 1

Используем [шаблон vite](https://vitejs.dev/guide/) для создания приложения:

```bash
# npm
npm create vite@latest
# yarn
yarn create vite
# or pnpm
pnpm create vite
```

Следуем инструкциям, называем проект, выбираем React и нужный язык

### Шаг 2

В папке с проектом устанавливаем VKUI и другие нужные библиотеки

```bash
# npm
npm i --save @vkontakte/vkui @vkontakte/icons @vkontakte/vk-bridge
# yarn
yarn add @vkontakte/vkui @vkontakte/icons @vkontakte/vk-bridge
# or pnpm
pnpm add @vkontakte/vkui @vkontakte/icons @vkontakte/vk-bridge
```

`@vkontakte/icons` — это наши иконки, которые напрямую используются в некоторых компонентах VKUI.
`@vkontakte/vk-bridge` необходима для интеграции в клиенты VK (она обеспечит корректные отступы на мобильных устройствах, свайпы, определение темы и тд).

### Шаг 3

В файле index.html необходимо добавить тег viewport для корректного отображения интерфейса на безрамочных смартфонах (подробнее см. в статье ["The Notch" and CSS](https://css-tricks.com/the-notch-and-css/)).

```diff
- <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+ <meta
+   name="viewport"
+   content="width=device-width, initial-scale=1, shrink-to-fit=no, user-scalable=no, viewport-fit=cover"
+ />
```

### Шаг 4

Собираем базовый app shell для VKUI

```ts
// src/main.js

import React from "react";
import ReactDOM from "react-dom/client";
import { AdaptivityProvider, ConfigProvider } from "@vkontakte/vkui";
import bridge from "@vkontakte/vk-bridge";
import App from "./App";
import "@vkontakte/vkui/dist/vkui.css";

ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    <ConfigProvider>
      <AdaptivityProvider>
        <App />
      </AdaptivityProvider>
    </ConfigProvider>
  </React.StrictMode>
);
```

```ts
// src/App.js

import {
  AppRoot,
  SplitLayout,
  SplitCol,
  View,
  Panel,
  PanelHeader,
  Header,
  Group,
  SimpleCell,
} from "@vkontakte/vkui";

function App() {
  return (
    <AppRoot>
      <SplitLayout header={<PanelHeader separator={false} />}>
        <SplitCol autoSpaced>
          <View activePanel="main">
            <Panel id="main">
              <PanelHeader>VKUI</PanelHeader>
              <Group header={<Header mode="secondary">Items</Header>}>
                <SimpleCell>Hello</SimpleCell>
                <SimpleCell>World</SimpleCell>
              </Group>
            </Panel>
          </View>
        </SplitCol>
      </SplitLayout>
    </AppRoot>
  );
}

export default App;
```

### Шаг 5

Для работы VK Mini Apps необходимо добавить вызов [VKWebAppInit](https://dev.vk.com/bridge/VKWebAppInit)(например в main.js), иначе приложение будет показывать бесконечную загрузку.

```js
// Init VK  Mini App
bridge.send("VKWebAppInit");
```

---

По итогу должен получится [такой проект](https://codesandbox.io/p/sandbox/vk-mini-apps-vite-eu1h7c)

**Reply by scffs** [reply]
У меня какая-то проблема с запуском на https, но это уже не проблема вк ui, спасибо за помощь

**Reply by scffs** [reply]
Можно ли связать с миниаппом без https? (т.е. на http сервере)
Проект собрался

**Reply by SevereCloud** [reply]
Нужен https. Можно использовать [vk-tunnel](https://dev.vk.com/tunnel)

**Reply by scffs** [reply]
На сайте получаю ошибку 1 (error 1)
Вроде всё делал по доке..

Странная запись ошибки, зачем писать `Error: Error:`
![изображение](https://user-images.githubusercontent.com/67464545/226928654-96bbf737-8d32-4448-9fc3-36f11fee3381.png)

![изображение](https://user-images.githubusercontent.com/67464545/226928839-41f96017-c85d-4e83-8053-c347d7e91e2c.png)
![изображение](https://user-images.githubusercontent.com/67464545/226928866-3703a5ea-898e-472c-ad88-e99e5c59ae38.png)

**Reply by scffs** [reply]
Изменил index.jsx на index.js, ошибка другая теперь

![изображение](https://user-images.githubusercontent.com/67464545/226930583-0d19f6ff-3d10-4b9c-8f12-00d49a54e777.png)

Параметры те же...
В настройки приложения ссылку добавил

**Reply by SevereCloud** [reply]
- `--insecure` не нужен, так как не используется https
- `--http-protocol` не нужен, так как vite работает http
- `--ws-protocol` не нужен, так как vite использует просто ws

**Reply by SevereCloud** [reply]
Команды `vk-tunnel --port=5173` должно хватить, чтобы запустить туннель.

<img width="1247" alt="image" src="https://user-images.githubusercontent.com/14944123/226931893-e9029c0d-1915-42a4-8b6e-5cdf729bbb7a.png">

**Reply by scffs** [reply]
Порт и хост же указывать как в вайте? Потому что ошибка всё ещё осталась, но уже другая

![изображение](https://user-images.githubusercontent.com/67464545/226932522-942294c2-de89-4a2a-a1bd-a5bdc18f9992.png)

**Reply by scffs** [reply]
> Команды `vk-tunnel --port=5173` должно хватить, чтобы запустить туннель.
> 
> <img alt="image" width="1247" src="https://user-images.githubusercontent.com/14944123/226931893-e9029c0d-1915-42a4-8b6e-5cdf729bbb7a.png">

помогло...спасибо....

**Reply by scffs** [reply]
С горе пополам сервер поднялся, но в консоли сыпятся ошибки..
При этом в самой ide всё окей
![изображение](https://user-images.githubusercontent.com/67464545/226934341-9d9fbb5c-caf8-459b-ac88-1f42f7ea9746.png)

Я, конечно, поищу решение, но буду рад очередной помощи, если это связано с vk tunnel

**Reply by scffs** [reply]
Починил.
Спасибо огромное, Даниил </3
